ros_ws/src/control/include/control/twist_mux.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file twist_mux.h
+ * @class TwistMux
+ * @brief A simple class which accepts multiple twist messages and outputs only one
+ *
+ * This class should act as a way to select which twist message is being sent to the robots motor controllers.  The
+ * message can come from multiple places including the teleop node, the estop node (built in to this class), or the
+ * autonomous node.  The twist mux will use joy messages to decide which twist message should be used at any given
+ * time.
+ */
+
+#ifndef CONTROL_INCLUDE_CONTROL_TWIST_MUX_H_
+#define CONTROL_INCLUDE_CONTROL_TWIST_MUX_H_
+
+#include <ros/ros.h>
+#include <sensor_msgs/Joy.h>
+#include <geometry_msgs/Twist.h>
+#include "constants.h"
+
+class TwistMux
+{
+private:
+    /**
+     * @brief An enumerator to describe the mode the robot is currently in
+     */
+    enum Mode
+    {
+        standby,        //!< The robot is not moving, the same as a software e-stop
+        teleop,         //!< The robot is currently being controlled by remote control
+        autonomous      //!< The robot is driving itself
+    };
```

Comment:
This (unscoped) enum will pollute the global namespace with the names of its members.

Please see [the ROS C++ Style Guide section on Enumerations](http://wiki.ros.org/CppStyleGuide#Enumerations) for the proper way to define enums without polluting the global namespace.